sql/2022/sustainability/script_count.sql

```diff
@@ -0,0 +1,26 @@
+#standardSQL
+# The percentage of scripts used inline v. external
+SELECT
+  client,
+  COUNT(0) AS total_scripts,
+  SUM(IF(script NOT LIKE '%src%', 1, 0)) AS inline_script,
+  SUM(IF(script LIKE '%src%', 1, 0)) AS external_script,
+  SUM(IF(script LIKE '%src%', 1, 0)) / COUNT(0) AS pct_external_script,
+  SUM(IF(script NOT LIKE '%src%', 1, 0)) / COUNT(0) AS pct_inline_script
+FROM
+  (
+    SELECT
+      client,
+      page,
+      url,
+      REGEXP_EXTRACT_ALL(LOWER(body), '(<script [^>]*)') AS scripts
+    FROM
+      `httparchive.almanac.summary_response_bodies`
```

Comment:
We REALY try to avoid this table if at all possible because of a few reasons:
1. It's MASSIVE and ridiculously costly to query. This makes it's use fairly limited outside of the almanac as most researchers won['t be able to reuse our queries. Though limiting it to just `firstHTML` does help with that.
2. It's the initial response delivered from the server. In this day and age JavaScript will add more script tags so we should look at the rendered HTML rather than the response body.
3. Regex is brittle as anything (e.g. we could have `<script data="<something>" src=...>` which would not work for your query).

You are probably better looking at the `$._javascript` of the body payload of the pages table. It gathers [this info](https://github.com/HTTPArchive/custom-metrics/blob/main/dist/javascript.js)

Here's an example query of how to query it: https://github.com/HTTPArchive/almanac.httparchive.org/blob/main/sql/2021/javascript/breakdown_of_scripts_using_async_defer_module_nomodule.sql